.github/workflows/benchmark.yml

```diff
@@ -70,18 +70,57 @@ jobs:
       run: |
         pytest tests
 
+  benchmarks:
+    name: Generator performance
+    runs-on: benchmark
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ['3.7', '3.8', '3.9', '3.10', '3.11']
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 1
+
+    - uses: chia-network/actions/setup-python@main
+      name: Install Python ${{ matrix.python-version }}
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Update pip
+      run: |
+          python -m pip install --upgrade pip
```

Comment:
I would probably do this inside the venv below  with the dependency installation.  I think this will be inherited into the env, but may as well just update it where we actually care.  Kind of counter to the comment there, _this_ would be a separate call two pip.  So you end up with two.  One to update install-related stuff like pip and maybe setuptools or wheel or build, and another to install every other package.